src/transformers/training_args.py

```diff
@@ -889,6 +914,11 @@ def should_log(self):
             else:
                 return self.process_index == 0
 
+    def get_node_log_level(self):
```

Comment:
@sgugger, I think I should have called this one differently, as it's not node-specific. 

Should we call it `get_process_log_level`?

Reply:
we can rename it during https://github.com/huggingface/transformers/issues/12295

Reply:
and I totally missed the docstring, so probably something like:

"""
Returns the log level to be used depending on whether this process is the main process of node 0, main process of node non-0, or a non-main process.

For the main process the log level defaults to ``logging.INFO`` unless overridden by ``log_level`` argument.

For the replica processes the log level defaults to ``logging.WARNING`` unless overridden by ``log_level_replica`` argument.

The choice between the main and replica process settings is made according to the return value of ``should_log``.
"""

Reply:
Yes, agree with all of this.

Reply:
implemented in https://github.com/huggingface/transformers/pull/12309